dyno-jedis/src/test/java/com/netflix/dyno/jedis/RedisAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+package com.netflix.dyno.jedis;
+
+import com.netflix.dyno.connectionpool.Host;
+import com.netflix.dyno.connectionpool.HostSupplier;
+import com.netflix.dyno.connectionpool.TokenMapSupplier;
+import com.netflix.dyno.connectionpool.impl.ConnectionPoolConfigurationImpl;
+import com.netflix.dyno.connectionpool.impl.lb.HostToken;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import redis.embedded.RedisServer;
+import redis.embedded.RedisServerBuilder;
+
+public class RedisAuthenticationIntegrationTest {
```

Comment:
It will be useful to also add another testcase showing failure to connect to redisServer (with Auth) and client (no auth).

Reply:
I had to drop from the `DynoJedisClient` down to the `JedisConnectionFactory` to get any kind of valuable unhappy path tests.  I've added [test methods](https://github.com/davidwadden/dyno/blob/redis-auth/dyno-jedis/src/test/java/com/netflix/dyno/jedis/RedisAuthenticationIntegrationTest.java#L90-L207) to cover these cases from a lower layer, with the following cases added:

- Connect failed
- Password required
- Invalid password